lib/internal/modules/package_json_reader.js

```diff
@@ -38,4 +38,14 @@ function read(jsonPath) {
   return result;
 }
 
-module.exports = { read };
+/**
+ * 
+ * @param {string} jsonPath 
+ * @param {string} srcPath 
+ */
+function clobber(jsonPath, srcPath) {
```

Comment:
How about just making this `set`, and then providing the direct JSON to set into the cache? Since `read` is called before the usage of `clobber` anyway. This then also opens the door to possible inline package configuration options.

Reply:
inline package configuration??

Reply:
Yeah! (eg `node app.js --pcfg.type=module` as a natural extension)

Reply:
ah, i was imagining just allowing a source text for the file since manually configuring multiple of them would be noise for the args parser

Reply:
Yeah I wouldn't want to speculate, suggestion was as much about read + clobber = read + read + write -> read + write being the simplification.